projects/cds/src/profiletag/model/profile-tag.model.ts

```diff
@@ -20,14 +22,51 @@ export interface ConsentReferenceEvent extends CustomEvent {
     consentReference: string;
   };
 }
+
 export interface DebugEvent extends CustomEvent {
   detail: {
     debug: boolean;
   };
 }
 
+export class NavigatedPushEvent {
+  event: ProfileTagPushEventNames.NAVIGATED;
+}
+
+export class ConsentChangedPushEvent {
+  event: ProfileTagPushEventNames.CONSENT_CHANGED;
+  granted: boolean;
+  constructor(granted: boolean) {
+    this.granted = granted;
+  }
+}
+
+export class CartChangedPushEvent {
+  event: ProfileTagPushEventNames.CART_CHANGED;
+  data: ProfileTagCart;
+  constructor(data: ProfileTagCart) {
+    this.data = data;
+  }
+}
+
+interface ProfileTagCart {
+  entries: OrderEntry[];
+  cart: Cart;
+}
+
+export type PushEvent =
+  | NavigatedPushEvent
+  | ConsentChangedPushEvent
+  | CartChangedPushEvent;
+
+export enum ProfileTagPushEventNames {
+  NAVIGATED = 'Navigated',
+  CONSENT_CHANGED = 'ConsentChanged',
+  CART_CHANGED = 'CartSnapshot',
+}
+
 export enum ProfileTagEventNames {
   LOADED = 'profiletag_loaded',
-  CONSENT_REFERENCE_CHANGED = 'profiletag_consentReferenceChanged',
+  CONSENT_REFERENCE_CHANGED = 'profiletag_consentReferenceLoaded',
```

Comment:
I noticed that the event is now called _consent reference loaded_. Should the enum _CONSENT_REFERENCE_CHANGED_ be updated to e.g. _CONSENT_REFERENCE_UPDATED_ as well?